resolver/src/lookup_ip.rs

```diff
@@ -71,6 +74,7 @@ pub struct InnerLookupIpFuture<C: ClientHandle + 'static> {
     names: Vec<Name>,
     strategy: LookupIpStrategy,
     future: Box<Future<Item = Lookup, Error = io::Error>>,
+    hosts: Arc<Hosts>,
```

Comment:
Should be an `Option<Arc<..>>` since it can be disabled in the opts?

Reply:
yes, `Option` is better